packages/panna-sdk/src/react/components/add-funds/transfer-from-wallet/transfer-select-token-step.tsx

```diff
@@ -0,0 +1,170 @@
+import { useEffect, useMemo, useState } from 'react';
+import { UseFormReturn } from 'react-hook-form';
+import { Label } from '@/components/ui/label';
+import { TokenBalance } from '@/mocks/token-balances';
+import { getEnvironmentChain, getSupportedTokens } from '@/utils';
+import { useExternalWallet, usePanna, useTokenBalances } from '../../../hooks';
+import { Card } from '../../ui/card';
+import { DialogHeader, DialogTitle } from '../../ui/dialog';
+import { useDialogStepper } from '../../ui/dialog-stepper';
+import { Skeleton } from '../../ui/skeleton';
+import { Typography } from '../../ui/typography';
+import { TransferFormData } from './schema';
+
+type TransferSelectTokenStepProps = {
+  form: UseFormReturn<TransferFormData>;
+};
+
+export function TransferSelectTokenStep({
+  form
+}: TransferSelectTokenStepProps) {
+  const { next } = useDialogStepper();
+  const { externalAddress } = useExternalWallet();
+  const { chainId } = usePanna();
+  const [selectedToken, setSelectedToken] = useState<
+    TransferFormData['tokenInfo'] | null
+  >(null);
+  const supportedTokens = getSupportedTokens(chainId);
+  const chainInitial = useMemo(() => {
+    const chain = getEnvironmentChain(chainId);
+    const chainName = chain?.name ?? '';
+
+    if (chainName.length > 0) {
+      return chainName.charAt(0).toUpperCase();
+    }
+
+    return 'N';
+  }, [chainId]);
```

Comment:
Should the suggested `chainInitial` changes in `transfer-amount-step` be made, they should also be updated here.